test/test_testing.py

```diff
@@ -5854,3 +5854,34 @@ def test_get_notices(self):
 class TestNotices(unittest.TestCase, _TestingPebbleClientMixin, PebbleNoticesMixin):
     def setUp(self):
         self.client = self.get_testing_client()
+
+
+class TestCloudSpec(unittest.TestCase):
+    def test_get_set_cloud_spec(self):
+        harness = ops.testing.Harness(EventRecorder, meta='name: myapp')
+        self.addCleanup(harness.cleanup)
+        cloud_spec_dict = {
+            'name': 'localhost',
+            'type': 'lxd',
+            'endpoint': 'https://127.0.0.1:8443',
+            'credential': {
+                'authtype': 'certificate',
+                'attrs': {
+                    'client-cert': 'foo',
+                    'client-key': 'bar',
+                    'server-cert': 'baz'
+                },
+            },
+        }
+        harness.set_cloud_spec(ops.model.CloudSpec.from_dict(cloud_spec_dict))
```

Comment:
Could use the `CloudSpec` constructor directly here with just the required fields (we've already tested all fields in `test_model.py`). Also, prefer `ops.CloudSpec` over `ops.model.CloudSpec`. So something like:

```python
harness.set_cloud_spec(ops.CloudSpec(name='localhost', type='lxd'))
```

Also, it seems to me this isn't the kind of test charmers would actually do. Why don't we basically reproduce the test from your docstring example earlier, which creates a little charm with on_start and sets a `self.spec` attribute that we assert on.